### DISCUSSION TITLE
Owner ID

### DISCUSSION THREAD

**Discussion by Skigamer13** [Rapptz/discord.py#6816] [Q&A] [answered]
I'm having trouble coding the bot to recognize me as the owner, this is the code I'm currently using "discord.ext.commands.Bot.owner_id(836449409226637313)
discord.ext.commands.bot.owner_is(discord.User)" but it keeps on saying 
"Traceback (most recent call last):
  File "main.py", line 8, in <module>
    discord.ext.commands.Bot.owner_id(836449409226637313)
AttributeError: type object 'Bot' has no attribute 'owner_id'"

**Comment by mesub7** [comment, marked as answer]
Hey there,

If you don't have a command which users the [`is_owner()`](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.is_owner) check, then you'll need to define `Bot.owner_id` yourself (`bot.owner_id = ...`).

Take care.

**Reply by Skigamer13** [reply]
How exactly do I define the owner id? I've tried multiple times with different ways of writing it out but I haven't been having much luck getting it to work.

**Reply by mesub7** [reply]
You would create your `commands.bot` instance and then define it afterwards:

```py
bot = commands.Bot(...) #options etc
bot.owner_id = ...
```

**Reply by Skigamer13** [reply]
What should I put in the parentheses for commands.Bot()? And is that all the code I need to identify the owner? Also, thank you for the help.

**Reply by 15696** [reply]
> What should I put in the parentheses for commands.Bot()?

What you already have.

**Reply by Skigamer13** [reply]
Alright, I looked through the documentation and I tried this code below, but it's giving the response that means the bot doesn't recognize an ID as that of the owner's. Does anybody know how to troubleshoot this part?

def is_owner():
  async def predicate(ctx):
    return ctx.author.id == 836449409226637313

    return commands.check(predicate)


@bot.command()
@commands.is_owner()
async def secretownercommand(ctx):
    await ctx.send('Hello Maker!')


@secretownercommand.error
async def secretownercommand_error(ctx, error):
    if isinstance(error, commands.CheckFailure):
        await ctx.send('Nothing to see here friend.')

**Reply by Skigamer13** [reply]
Never mind, I double-checked and I figured out I accidentally used the wrong type of ID, so I corrected it and now everything works! Thanks for all the help everyone!
